{
  "origin": "codeshovel",
  "repositoryName": "hadoop",
  "repositoryPath": "/home/shaiful/research/codeshovel/codeshovel-projects/hadoop/.git",
  "startCommitName": "HEAD",
  "sourceFileName": "ClientRMService.java",
  "functionName": "getApplicationAttemptReport",
  "functionId": "getApplicationAttemptReport___request-GetApplicationAttemptReportRequest",
  "sourceFilePath": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ClientRMService.java",
  "functionStartLine": 439,
  "functionEndLine": 469,
  "numCommitsSeen": 128,
  "timeTaken": 3691,
  "changeHistory": [
    "cffea251be4b73ca16e5e11f0be2d22651330f73",
    "e0bb867c3fa638c9f689ee0b044b400481cf02b5",
    "1bbd023275db535ab80fcb60e022151e9679d468",
    "18b2507edaac991e3ed68d2f27eb96f6882137b9",
    "6af044bb75fe3ddf184841483cce5e222ad1853a"
  ],
  "changeHistoryShort": {
    "cffea251be4b73ca16e5e11f0be2d22651330f73": "Ybodychange",
    "e0bb867c3fa638c9f689ee0b044b400481cf02b5": "Ybodychange",
    "1bbd023275db535ab80fcb60e022151e9679d468": "Ybodychange",
    "18b2507edaac991e3ed68d2f27eb96f6882137b9": "Ybodychange",
    "6af044bb75fe3ddf184841483cce5e222ad1853a": "Yintroduced"
  },
  "changeHistoryDetails": {
    "cffea251be4b73ca16e5e11f0be2d22651330f73": {
      "type": "Ybodychange",
      "commitMessage": "YARN-5956. Refactor ClientRMService for unify error handling across apis. Contributed by Kai Sasaki.\n",
      "commitDate": "25/03/17 9:56 AM",
      "commitName": "cffea251be4b73ca16e5e11f0be2d22651330f73",
      "commitAuthor": "Sunil G",
      "commitDateOld": "28/02/17 11:04 AM",
      "commitNameOld": "e0bb867c3fa638c9f689ee0b044b400481cf02b5",
      "commitAuthorOld": "Xuan",
      "daysBetweenCommits": 24.91,
      "commitsBetweenForRepo": 150,
      "commitsBetweenForFile": 1,
      "diff": "@@ -1,41 +1,31 @@\n   public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n       GetApplicationAttemptReportRequest request) throws YarnException,\n       IOException {\n+    ApplicationId applicationId\n+        \u003d request.getApplicationAttemptId().getApplicationId();\n     ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n-    UserGroupInformation callerUGI;\n-    try {\n-      callerUGI \u003d UserGroupInformation.getCurrentUser();\n-    } catch (IOException ie) {\n-      LOG.info(\"Error getting UGI \", ie);\n-      throw RPCUtil.getRemoteException(ie);\n-    }\n-    RMApp application \u003d this.rmContext.getRMApps().get(\n-        appAttemptId.getApplicationId());\n-    if (application \u003d\u003d null) {\n-      // If the RM doesn\u0027t have the application, throw\n-      // ApplicationNotFoundException and let client to handle.\n-      throw new ApplicationNotFoundException(\"Application with id \u0027\"\n-          + request.getApplicationAttemptId().getApplicationId()\n-          + \"\u0027 doesn\u0027t exist in RM. Please check that the job \"\n-          + \"submission was successful.\");\n-    }\n+    UserGroupInformation callerUGI \u003d getCallerUgi(applicationId,\n+        AuditConstants.GET_APP_ATTEMPT_REPORT);\n+    RMApp application \u003d verifyUserAccessForRMApp(applicationId, callerUGI,\n+        AuditConstants.GET_APP_ATTEMPT_REPORT, ApplicationAccessType.VIEW_APP,\n+        false);\n \n     boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n         ApplicationAccessType.VIEW_APP, application);\n     GetApplicationAttemptReportResponse response \u003d null;\n     if (allowAccess) {\n       RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n       if (appAttempt \u003d\u003d null) {\n         throw new ApplicationAttemptNotFoundException(\n             \"ApplicationAttempt with id \u0027\" + appAttemptId +\n             \"\u0027 doesn\u0027t exist in RM.\");\n       }\n       ApplicationAttemptReport attemptReport \u003d appAttempt\n           .createApplicationAttemptReport();\n       response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n     }else{\n       throw new YarnException(\"User \" + callerUGI.getShortUserName()\n           + \" does not have privilege to see this attempt \" + appAttemptId);\n     }\n     return response;\n   }\n\\ No newline at end of file\n",
      "actualSource": "  public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n      GetApplicationAttemptReportRequest request) throws YarnException,\n      IOException {\n    ApplicationId applicationId\n        \u003d request.getApplicationAttemptId().getApplicationId();\n    ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n    UserGroupInformation callerUGI \u003d getCallerUgi(applicationId,\n        AuditConstants.GET_APP_ATTEMPT_REPORT);\n    RMApp application \u003d verifyUserAccessForRMApp(applicationId, callerUGI,\n        AuditConstants.GET_APP_ATTEMPT_REPORT, ApplicationAccessType.VIEW_APP,\n        false);\n\n    boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n        ApplicationAccessType.VIEW_APP, application);\n    GetApplicationAttemptReportResponse response \u003d null;\n    if (allowAccess) {\n      RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n      if (appAttempt \u003d\u003d null) {\n        throw new ApplicationAttemptNotFoundException(\n            \"ApplicationAttempt with id \u0027\" + appAttemptId +\n            \"\u0027 doesn\u0027t exist in RM.\");\n      }\n      ApplicationAttemptReport attemptReport \u003d appAttempt\n          .createApplicationAttemptReport();\n      response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n    }else{\n      throw new YarnException(\"User \" + callerUGI.getShortUserName()\n          + \" does not have privilege to see this attempt \" + appAttemptId);\n    }\n    return response;\n  }",
      "path": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ClientRMService.java",
      "extendedDetails": {}
    },
    "e0bb867c3fa638c9f689ee0b044b400481cf02b5": {
      "type": "Ybodychange",
      "commitMessage": "YARN-6189: Improve application status log message when RM restarted when\napp is in NEW state. Contributed by Junping Du\n",
      "commitDate": "28/02/17 11:04 AM",
      "commitName": "e0bb867c3fa638c9f689ee0b044b400481cf02b5",
      "commitAuthor": "Xuan",
      "commitDateOld": "23/02/17 12:49 AM",
      "commitNameOld": "a207aa9930e7ee4f10228e2db4b4e733794eb8ea",
      "commitAuthorOld": "Sunil G",
      "daysBetweenCommits": 5.43,
      "commitsBetweenForRepo": 34,
      "commitsBetweenForFile": 1,
      "diff": "@@ -1,40 +1,41 @@\n   public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n       GetApplicationAttemptReportRequest request) throws YarnException,\n       IOException {\n     ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n     UserGroupInformation callerUGI;\n     try {\n       callerUGI \u003d UserGroupInformation.getCurrentUser();\n     } catch (IOException ie) {\n       LOG.info(\"Error getting UGI \", ie);\n       throw RPCUtil.getRemoteException(ie);\n     }\n     RMApp application \u003d this.rmContext.getRMApps().get(\n         appAttemptId.getApplicationId());\n     if (application \u003d\u003d null) {\n       // If the RM doesn\u0027t have the application, throw\n       // ApplicationNotFoundException and let client to handle.\n       throw new ApplicationNotFoundException(\"Application with id \u0027\"\n           + request.getApplicationAttemptId().getApplicationId()\n-          + \"\u0027 doesn\u0027t exist in RM.\");\n+          + \"\u0027 doesn\u0027t exist in RM. Please check that the job \"\n+          + \"submission was successful.\");\n     }\n \n     boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n         ApplicationAccessType.VIEW_APP, application);\n     GetApplicationAttemptReportResponse response \u003d null;\n     if (allowAccess) {\n       RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n       if (appAttempt \u003d\u003d null) {\n         throw new ApplicationAttemptNotFoundException(\n             \"ApplicationAttempt with id \u0027\" + appAttemptId +\n             \"\u0027 doesn\u0027t exist in RM.\");\n       }\n       ApplicationAttemptReport attemptReport \u003d appAttempt\n           .createApplicationAttemptReport();\n       response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n     }else{\n       throw new YarnException(\"User \" + callerUGI.getShortUserName()\n           + \" does not have privilege to see this attempt \" + appAttemptId);\n     }\n     return response;\n   }\n\\ No newline at end of file\n",
      "actualSource": "  public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n      GetApplicationAttemptReportRequest request) throws YarnException,\n      IOException {\n    ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n    UserGroupInformation callerUGI;\n    try {\n      callerUGI \u003d UserGroupInformation.getCurrentUser();\n    } catch (IOException ie) {\n      LOG.info(\"Error getting UGI \", ie);\n      throw RPCUtil.getRemoteException(ie);\n    }\n    RMApp application \u003d this.rmContext.getRMApps().get(\n        appAttemptId.getApplicationId());\n    if (application \u003d\u003d null) {\n      // If the RM doesn\u0027t have the application, throw\n      // ApplicationNotFoundException and let client to handle.\n      throw new ApplicationNotFoundException(\"Application with id \u0027\"\n          + request.getApplicationAttemptId().getApplicationId()\n          + \"\u0027 doesn\u0027t exist in RM. Please check that the job \"\n          + \"submission was successful.\");\n    }\n\n    boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n        ApplicationAccessType.VIEW_APP, application);\n    GetApplicationAttemptReportResponse response \u003d null;\n    if (allowAccess) {\n      RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n      if (appAttempt \u003d\u003d null) {\n        throw new ApplicationAttemptNotFoundException(\n            \"ApplicationAttempt with id \u0027\" + appAttemptId +\n            \"\u0027 doesn\u0027t exist in RM.\");\n      }\n      ApplicationAttemptReport attemptReport \u003d appAttempt\n          .createApplicationAttemptReport();\n      response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n    }else{\n      throw new YarnException(\"User \" + callerUGI.getShortUserName()\n          + \" does not have privilege to see this attempt \" + appAttemptId);\n    }\n    return response;\n  }",
      "path": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ClientRMService.java",
      "extendedDetails": {}
    },
    "1bbd023275db535ab80fcb60e022151e9679d468": {
      "type": "Ybodychange",
      "commitMessage": "YARN-5962. Spelling errors in logging and exceptions for resource manager code (gsohn via rkanter)\n",
      "commitDate": "27/12/16 1:35 PM",
      "commitName": "1bbd023275db535ab80fcb60e022151e9679d468",
      "commitAuthor": "Robert Kanter",
      "commitDateOld": "13/12/16 8:49 PM",
      "commitNameOld": "ada876cd1d22b61f237603cf339bbed65285dab8",
      "commitAuthorOld": "Jian He",
      "daysBetweenCommits": 13.7,
      "commitsBetweenForRepo": 53,
      "commitsBetweenForFile": 1,
      "diff": "@@ -1,40 +1,40 @@\n   public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n       GetApplicationAttemptReportRequest request) throws YarnException,\n       IOException {\n     ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n     UserGroupInformation callerUGI;\n     try {\n       callerUGI \u003d UserGroupInformation.getCurrentUser();\n     } catch (IOException ie) {\n       LOG.info(\"Error getting UGI \", ie);\n       throw RPCUtil.getRemoteException(ie);\n     }\n     RMApp application \u003d this.rmContext.getRMApps().get(\n         appAttemptId.getApplicationId());\n     if (application \u003d\u003d null) {\n       // If the RM doesn\u0027t have the application, throw\n       // ApplicationNotFoundException and let client to handle.\n       throw new ApplicationNotFoundException(\"Application with id \u0027\"\n           + request.getApplicationAttemptId().getApplicationId()\n           + \"\u0027 doesn\u0027t exist in RM.\");\n     }\n \n     boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n         ApplicationAccessType.VIEW_APP, application);\n     GetApplicationAttemptReportResponse response \u003d null;\n     if (allowAccess) {\n       RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n       if (appAttempt \u003d\u003d null) {\n         throw new ApplicationAttemptNotFoundException(\n             \"ApplicationAttempt with id \u0027\" + appAttemptId +\n             \"\u0027 doesn\u0027t exist in RM.\");\n       }\n       ApplicationAttemptReport attemptReport \u003d appAttempt\n           .createApplicationAttemptReport();\n       response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n     }else{\n       throw new YarnException(\"User \" + callerUGI.getShortUserName()\n-          + \" does not have privilage to see this attempt \" + appAttemptId);\n+          + \" does not have privilege to see this attempt \" + appAttemptId);\n     }\n     return response;\n   }\n\\ No newline at end of file\n",
      "actualSource": "  public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n      GetApplicationAttemptReportRequest request) throws YarnException,\n      IOException {\n    ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n    UserGroupInformation callerUGI;\n    try {\n      callerUGI \u003d UserGroupInformation.getCurrentUser();\n    } catch (IOException ie) {\n      LOG.info(\"Error getting UGI \", ie);\n      throw RPCUtil.getRemoteException(ie);\n    }\n    RMApp application \u003d this.rmContext.getRMApps().get(\n        appAttemptId.getApplicationId());\n    if (application \u003d\u003d null) {\n      // If the RM doesn\u0027t have the application, throw\n      // ApplicationNotFoundException and let client to handle.\n      throw new ApplicationNotFoundException(\"Application with id \u0027\"\n          + request.getApplicationAttemptId().getApplicationId()\n          + \"\u0027 doesn\u0027t exist in RM.\");\n    }\n\n    boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n        ApplicationAccessType.VIEW_APP, application);\n    GetApplicationAttemptReportResponse response \u003d null;\n    if (allowAccess) {\n      RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n      if (appAttempt \u003d\u003d null) {\n        throw new ApplicationAttemptNotFoundException(\n            \"ApplicationAttempt with id \u0027\" + appAttemptId +\n            \"\u0027 doesn\u0027t exist in RM.\");\n      }\n      ApplicationAttemptReport attemptReport \u003d appAttempt\n          .createApplicationAttemptReport();\n      response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n    }else{\n      throw new YarnException(\"User \" + callerUGI.getShortUserName()\n          + \" does not have privilege to see this attempt \" + appAttemptId);\n    }\n    return response;\n  }",
      "path": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ClientRMService.java",
      "extendedDetails": {}
    },
    "18b2507edaac991e3ed68d2f27eb96f6882137b9": {
      "type": "Ybodychange",
      "commitMessage": "YARN-1904. Ensure exceptions thrown in ClientRMService \u0026 ApplicationHistoryClientService are uniform when application-attempt is not found. Contributed by Zhijie Shen.\n",
      "commitDate": "05/02/15 11:48 PM",
      "commitName": "18b2507edaac991e3ed68d2f27eb96f6882137b9",
      "commitAuthor": "Arun C. Murthy",
      "commitDateOld": "30/12/14 5:15 PM",
      "commitNameOld": "e2351c7ae24cea9b217af4174512d279c55e8efd",
      "commitAuthorOld": "Jian He",
      "daysBetweenCommits": 37.27,
      "commitsBetweenForRepo": 260,
      "commitsBetweenForFile": 1,
      "diff": "@@ -1,39 +1,40 @@\n   public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n       GetApplicationAttemptReportRequest request) throws YarnException,\n       IOException {\n     ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n     UserGroupInformation callerUGI;\n     try {\n       callerUGI \u003d UserGroupInformation.getCurrentUser();\n     } catch (IOException ie) {\n       LOG.info(\"Error getting UGI \", ie);\n       throw RPCUtil.getRemoteException(ie);\n     }\n     RMApp application \u003d this.rmContext.getRMApps().get(\n         appAttemptId.getApplicationId());\n     if (application \u003d\u003d null) {\n       // If the RM doesn\u0027t have the application, throw\n       // ApplicationNotFoundException and let client to handle.\n       throw new ApplicationNotFoundException(\"Application with id \u0027\"\n           + request.getApplicationAttemptId().getApplicationId()\n           + \"\u0027 doesn\u0027t exist in RM.\");\n     }\n \n     boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n         ApplicationAccessType.VIEW_APP, application);\n     GetApplicationAttemptReportResponse response \u003d null;\n     if (allowAccess) {\n       RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n       if (appAttempt \u003d\u003d null) {\n-        throw new ApplicationAttemptNotFoundException(\"ApplicationAttempt \"\n-            + appAttemptId + \" Not Found in RM\");\n+        throw new ApplicationAttemptNotFoundException(\n+            \"ApplicationAttempt with id \u0027\" + appAttemptId +\n+            \"\u0027 doesn\u0027t exist in RM.\");\n       }\n       ApplicationAttemptReport attemptReport \u003d appAttempt\n           .createApplicationAttemptReport();\n       response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n     }else{\n       throw new YarnException(\"User \" + callerUGI.getShortUserName()\n           + \" does not have privilage to see this attempt \" + appAttemptId);\n     }\n     return response;\n   }\n\\ No newline at end of file\n",
      "actualSource": "  public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n      GetApplicationAttemptReportRequest request) throws YarnException,\n      IOException {\n    ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n    UserGroupInformation callerUGI;\n    try {\n      callerUGI \u003d UserGroupInformation.getCurrentUser();\n    } catch (IOException ie) {\n      LOG.info(\"Error getting UGI \", ie);\n      throw RPCUtil.getRemoteException(ie);\n    }\n    RMApp application \u003d this.rmContext.getRMApps().get(\n        appAttemptId.getApplicationId());\n    if (application \u003d\u003d null) {\n      // If the RM doesn\u0027t have the application, throw\n      // ApplicationNotFoundException and let client to handle.\n      throw new ApplicationNotFoundException(\"Application with id \u0027\"\n          + request.getApplicationAttemptId().getApplicationId()\n          + \"\u0027 doesn\u0027t exist in RM.\");\n    }\n\n    boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n        ApplicationAccessType.VIEW_APP, application);\n    GetApplicationAttemptReportResponse response \u003d null;\n    if (allowAccess) {\n      RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n      if (appAttempt \u003d\u003d null) {\n        throw new ApplicationAttemptNotFoundException(\n            \"ApplicationAttempt with id \u0027\" + appAttemptId +\n            \"\u0027 doesn\u0027t exist in RM.\");\n      }\n      ApplicationAttemptReport attemptReport \u003d appAttempt\n          .createApplicationAttemptReport();\n      response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n    }else{\n      throw new YarnException(\"User \" + callerUGI.getShortUserName()\n          + \" does not have privilage to see this attempt \" + appAttemptId);\n    }\n    return response;\n  }",
      "path": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ClientRMService.java",
      "extendedDetails": {}
    },
    "6af044bb75fe3ddf184841483cce5e222ad1853a": {
      "type": "Yintroduced",
      "commitMessage": "YARN-1389. Made ApplicationClientProtocol and ApplicationHistoryProtocol expose analogous getApplication(s)/Attempt(s)/Container(s) APIs. Contributed by Mayank Bansal.\n\n\ngit-svn-id: https://svn.apache.org/repos/asf/hadoop/common/trunk@1577052 13f79535-47bb-0310-9956-ffa450edef68\n",
      "commitDate": "13/03/14 12:42 AM",
      "commitName": "6af044bb75fe3ddf184841483cce5e222ad1853a",
      "commitAuthor": "Zhijie Shen",
      "diff": "@@ -0,0 +1,39 @@\n+  public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n+      GetApplicationAttemptReportRequest request) throws YarnException,\n+      IOException {\n+    ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n+    UserGroupInformation callerUGI;\n+    try {\n+      callerUGI \u003d UserGroupInformation.getCurrentUser();\n+    } catch (IOException ie) {\n+      LOG.info(\"Error getting UGI \", ie);\n+      throw RPCUtil.getRemoteException(ie);\n+    }\n+    RMApp application \u003d this.rmContext.getRMApps().get(\n+        appAttemptId.getApplicationId());\n+    if (application \u003d\u003d null) {\n+      // If the RM doesn\u0027t have the application, throw\n+      // ApplicationNotFoundException and let client to handle.\n+      throw new ApplicationNotFoundException(\"Application with id \u0027\"\n+          + request.getApplicationAttemptId().getApplicationId()\n+          + \"\u0027 doesn\u0027t exist in RM.\");\n+    }\n+\n+    boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n+        ApplicationAccessType.VIEW_APP, application);\n+    GetApplicationAttemptReportResponse response \u003d null;\n+    if (allowAccess) {\n+      RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n+      if (appAttempt \u003d\u003d null) {\n+        throw new ApplicationAttemptNotFoundException(\"ApplicationAttempt \"\n+            + appAttemptId + \" Not Found in RM\");\n+      }\n+      ApplicationAttemptReport attemptReport \u003d appAttempt\n+          .createApplicationAttemptReport();\n+      response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n+    }else{\n+      throw new YarnException(\"User \" + callerUGI.getShortUserName()\n+          + \" does not have privilage to see this attempt \" + appAttemptId);\n+    }\n+    return response;\n+  }\n\\ No newline at end of file\n",
      "actualSource": "  public GetApplicationAttemptReportResponse getApplicationAttemptReport(\n      GetApplicationAttemptReportRequest request) throws YarnException,\n      IOException {\n    ApplicationAttemptId appAttemptId \u003d request.getApplicationAttemptId();\n    UserGroupInformation callerUGI;\n    try {\n      callerUGI \u003d UserGroupInformation.getCurrentUser();\n    } catch (IOException ie) {\n      LOG.info(\"Error getting UGI \", ie);\n      throw RPCUtil.getRemoteException(ie);\n    }\n    RMApp application \u003d this.rmContext.getRMApps().get(\n        appAttemptId.getApplicationId());\n    if (application \u003d\u003d null) {\n      // If the RM doesn\u0027t have the application, throw\n      // ApplicationNotFoundException and let client to handle.\n      throw new ApplicationNotFoundException(\"Application with id \u0027\"\n          + request.getApplicationAttemptId().getApplicationId()\n          + \"\u0027 doesn\u0027t exist in RM.\");\n    }\n\n    boolean allowAccess \u003d checkAccess(callerUGI, application.getUser(),\n        ApplicationAccessType.VIEW_APP, application);\n    GetApplicationAttemptReportResponse response \u003d null;\n    if (allowAccess) {\n      RMAppAttempt appAttempt \u003d application.getAppAttempts().get(appAttemptId);\n      if (appAttempt \u003d\u003d null) {\n        throw new ApplicationAttemptNotFoundException(\"ApplicationAttempt \"\n            + appAttemptId + \" Not Found in RM\");\n      }\n      ApplicationAttemptReport attemptReport \u003d appAttempt\n          .createApplicationAttemptReport();\n      response \u003d GetApplicationAttemptReportResponse.newInstance(attemptReport);\n    }else{\n      throw new YarnException(\"User \" + callerUGI.getShortUserName()\n          + \" does not have privilage to see this attempt \" + appAttemptId);\n    }\n    return response;\n  }",
      "path": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ClientRMService.java"
    }
  }
}